### PR TITLE
Adds "chef analyze capture NODE"

### PR DESCRIPTION
**Pull request by marcparadise**
This adds an initial `capture` command to `chef analyze`.

`chef analyze capture`  accepts a node as an argument and creates a chef repo
that contains:

* node object
* required cookbooks
* roles
* environments

capture_068
capture_065

This scope of this change extends to creating and populating the chef repository,
but does not include configuring a node.

## Related
closes chef/chef-workstation#1017

Signed-off-by: Marc A. Paradise <marc.paradise@gmail.com>

### CODE REVIEW

**Comment by marcparadise** [comment]
Note: not ready for merge, I meant to submit as draft.

File: cmd/capture.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		Use:   "capture NODE-NAME",
+		Short: "Capture a node's state into a local chef-repo",
+		Args:  cobra.ExactArgs(1),
+		Long: `Captures a node's state as a local chef-repo, which
+can then be used to converge locally.`,
+		RunE: func(_ *cobra.Command, args []string) error {
+			// TODO - support multiple node names?
+			creds, err := credentials.FromViper(
+				// TODO this isn't just reportsflags any longer.
+				reportsFlags.profile,
```
Comment (tyler-ball) [inline cmd/capture.go]: Seems like we have some refactors to create a new `capture` module and extract shared logic to a different module, but I agreed that can be another PR

File: cmd/root.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ var (
 		Use:   dist.Exec,
 		Short: fmt.Sprintf("A CLI to analyze artifacts from a %s", dist.ServerProduct),
 		Long: fmt.Sprintf(`Analyze your %s artifacts to understand the effort to upgrade
-your infrastructure by generating reports, automatically fixing violations
+your infrastructure by generating eports, automatically fixing violations
```
Comment (marcparadise) [inline cmd/root.go]: fix

File: pkg/reporting/capture_test.go [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+	nc := subject.NewNodeCapture("",
+		"",
+		NodeMock{Error: nil, Node: expectedNode},
+		RoleMock{},
+		EnvMock{},
+		CookbookMock{},
+		&writer,
+	)
+	actualNode, err := nc.CaptureNodeObject()
+	assert.Equal(t, &expectedNode, actualNode)
```
Comment (marcparadise) [inline pkg/reporting/capture_test.go]: asset.Equal(t, &expectedNode, writer.ReceivedObject)

File: pkg/reporting/client.go [hunk 4]
```diff
@@ -32,3 +32,13 @@ type CookbookInterface interface {
 type SearchInterface interface {
 	PartialExec(idx, statement string, params map[string]interface{}) (res chef.SearchResult, err error)
 }
+
+type NodesInterface interface {
+	Get(name string) (chef.Node, error)
```
Comment (tyler-ball) [inline pkg/reporting/client.go]: Any reason the other interfaces return a reference and this returns an object?

File: pkg/reporting/object_writer.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+func (ow *ObjectWriter) WriteEnvironment(env *chef.Environment) error {
+	return ow.save("environments", env.Name, env)
+}
+
+func (ow *ObjectWriter) WriteNode(node *chef.Node) error {
+	return ow.save("nodes", node.Name, node)
+}
+
+func (ow *ObjectWriter) save(objGroupingName string, objName string, object interface{}) error {
```
Comment (tyler-ball) [inline pkg/reporting/object_writer.go]: From the interactive review: we do not currently have full test coverage on this file. We could pass in an `os` object that we mock, but then that object needs to get passed all the way down the calling chain.

@marcparadise What if we wrote an integration test for this instead of a unit test? That would at least test the happy path.

Comment (marcparadise) [reply pkg/reporting/object_writer.go]: I'm not sure we can - it needs a chef server environment right? 

Actually, that might be a new card - configure chef-analyze integration tests to launch an instance of chef-zero.  Then all kinds of things will be possible. 

After accepting that I couldn't get around testing that golang's file writing and checking code is working in a unit test, I added a happy-path test to verify that this function writes JSON content to the correct place, and that we can read it back in and have it be the thing we expect.

File: pkg/reporting/client.go (hunk 4 above)
Comment (marcparadise) [reply pkg/reporting/client.go]: The interface was made to match the [original function](https://github.com/go-chef/chef/blob/master/node.go#L50-L54) -  in this case,  `chef.Node.Get` returns an a struct instead of a pointer.

Comment (marcparadise) [reply pkg/reporting/client.go]: That bugs me every time I see it too :D

File: pkg/reporting/object_writer.go (hunk 5 above)
Comment (tyler-ball) [reply pkg/reporting/object_writer.go]: For this PR you did exactly what I was thinking - happy path test that JSON gets written out correctly.
